spec.md

```diff
@@ -1004,10 +1004,15 @@ For success responses, the following fields are defined:
 | dashboard_url | string | The URL of a web-based management user interface for the Service Instance; we refer to this as a service dashboard. The URL MUST contain enough information for the dashboard to identify the resource being accessed (`9189kdfsk0vfnku` in the example below). Note: a Service Broker that wishes to return `dashboard_url` for a Service Instance MUST return it with the initial response to the provision request, even if the service is provisioned asynchronously. If present, MUST be a string or null. |
 | operation | string | For asynchronous responses, Service Brokers MAY return an identifier representing the operation. The value of this field MUST be provided by the Platform with requests to the [Polling Last Operation for Service Instances](#polling-last-operation-for-service-instances) endpoint in a percent-encoded query parameter. If present, MAY be null, and MUST NOT contain more than 10,000 characters. |
 | metadata | [ServiceInstanceMetadata](#service-instance-metadata) object | An OPTIONAL object containing metadata for the Service Instance. |
+| attributes | object | An OPTIONAL object containing attributes generated by the Broker, attributes will be available in the response body when fetching information about a Service Instance. |
```

Comment:
Do you imagine returned attributes would impact platform behaviour changes? If not we could use the existing `metadata.labels` functionality, if the data is ok to be exposed by users.

If `metadata.labels` isn't suitable then I suggest we still key attributes under `metadata` and add some text on when it makes sense to use metadata.attributes vs metadata.labels.

Reply:
@Samze In the use-case mentioned in this PR the platform would receive a piece of important information in this field to complete the network handshake, this is the case for AWS PrivateLinks and the information we are sending is the `aws_service_name` and I imagine this being used by services that require a second step using information generated by the broker. I believe we could use metadata for this and we would have to add the `object` type to the definition since `metadata.labels` is not suitable in my opinion. Is metadata a better place for this information? I tried to avoid sticking important information as `metadata` and I am concerned that it will mix important information with `required` information.

Reply:
@Samze I was just looking at how we return `metadata` and looks like we never return as part of the `ServiceInstanceResource`, metadata is only present in the response of the provisioning. We could use metadata but then I believe we should normalize and include it in the response of both the resource and the provisioning.

Reply:
> ...a piece of important information in this field to complete the network handshake, this is the case for AWS PrivateLinks and the information we are sending is the aws_service_name and I imagine this being used by services that require a second step using information generated by the broker. 

Makes sense that `metadata.labels` isn't suitable if this is non-user facing information. I would be in favour of still placing it under `metadata` with a different key, seems redundant to have two places for this "broker" additional information,.

> @Samze I was just looking at how we return `metadata` and looks like we never return as part of the `ServiceInstanceResource`, metadata is only present in the response of the provisioning. We could use metadata but then I believe we should normalize and include it in the response of both the resource and the provisioning.

Goodpoint, I think `metadata` makes sense to be returned as part of GET on the instance resource. It is at the same level

Reply:
@Samze I believe I completed moving everything to `metadata` and I added a section explaining what attributes are supposed to mean in this context and I believe this is ready.